emu/src/cpu/instruction.rs

```diff
@@ -4,34 +4,66 @@ use logger::log;
 
 use crate::bitwise::Bits;
 
+use super::condition::Condition;
+
 #[derive(Debug, PartialEq, Eq)]
 pub enum ArmModeInstruction {
     DataProcessing,
     Multiply,
     MultiplyLong,
     SingleDataSwap,
-    BranchAndExchange,
+    BranchAndExchange(Condition, usize),
     HalfwordDataTransferRegisterOffset,
     HalfwordDataTransferImmediateOffset,
     SingleDataTransfer,
     Undefined,
     BlockDataTransfer,
-    Branch,
+    Branch(Condition, bool, u32),
     CoprocessorDataTransfer,
     CoprocessorDataOperation,
     CoprocessorRegisterTrasfer,
     SoftwareInterrupt,
 }
+impl ArmModeInstruction {
+    pub(crate) fn disassembler(&self) -> String {
+        match self {
+            Self::DataProcessing => "".to_owned(),
+            Self::Multiply => "".to_owned(),
+            Self::MultiplyLong => "".to_owned(),
+            Self::SingleDataSwap => "".to_owned(),
+            Self::BranchAndExchange(_, reg) => format!("bx R{}", reg),
+            Self::HalfwordDataTransferRegisterOffset => "".to_owned(),
+            Self::HalfwordDataTransferImmediateOffset => "".to_owned(),
+            Self::SingleDataTransfer => "".to_owned(),
+            Self::Undefined => "".to_owned(),
+            Self::BlockDataTransfer => "".to_owned(),
+            Self::Branch(cond, is_link, offset) => {
```

Comment:
The assembler syntax for branch/branchlink is `B{L}{cond} <expression>` where expression is the address where we want to branch to, not the offset. The offset is then encoded in the opcode but the asm syntax shows the final address

Reply:
Ok, I can't do this without changing many things... Because for having address I need to access to `PC`

Reply:
Uhm I get it but if we want to do a disassembler I think we should stick to the official format